File: cpp File/5585/5585.cpp
#include <iostream>

using namespace std;

int coin[6] = { 500, 100, 50, 10, 5, 1 };

int cost;
int cnt = 0;

int main()
{
	ios_base::sync_with_stdio(0);
	cin.tie(0);
	cout.tie(0);

	cin >> cost;

	cost = 1000 - cost;

	for (int i = 0; i < 6; i++)
	{
		if (cost >= coin[i])
		{
			cnt += (cost / coin[i]);
			cost %= coin[i];
		}
	}

	cout << cnt << "\n";

	return 0;
}

File: kotlin File/5585.kt
val coin = arrayOf(500, 100, 50, 10, 5, 1)

fun main() {
    var a = readLine()!!.toInt()

    a = 1000 - a

    var cnt = 0

    for (i in 0..5) {
        if (a / coin[i] > 0) {
            cnt += a / coin[i]

            a %= coin[i]
        }
    }

    println(cnt)
}
